### ISSUE TITLE
SHM-2054: Investigate: Generate list of files that need linting fix so we know the totals.

### ISSUE THREAD

**Issue by chrischabot** [shardeum-mirror/shardeum#39]
field | value
| - | - |
url | https://linear.app/shm/issue/SHM-2054/investigate-generate-list-of-files-that-need-linting-fix-so-we-know
author | @
date | Fri Aug  4 05:52:06 BST 2023
state | Done
project | Linting of Explorer and Collector
priority | High
assignee | @
labels | 
related | 
parent | 
children | 
PRs | 
attachments | 

Generate list of files that need linting fixes and remove from ignore patterns.

Andrew wants us to start with explorer, explorer collector as we may be sharing that code soon. We will use these lists to generate individual linting fix tasks and split it out for people to take when they need work.

# Findings and Suggestions

Here are the contents of “ignore patterns” for each project. If a project doesn’t have linting set up or needs some linting admin work then I have set it up and added the MR. The MR should be completed before the other linting work can be done

* [ ] thing1
* [ ] thing2
* [ ] thing3

### Explorer

1. "collector.ts”
2. 4 errors that have slipped in since the rest of the linting work was done that need fixed

### JSON RPC Server (Has been merged)

mr: [https://gitlab.com/shardeum/json-rpc-server/-/merge_requests/43](https://gitlab.com/shardeum/json-rpc-server/-/merge_requests/43)

1. "api.ts"
2. "utils.ts"
3. "config.ts"
4. "logger.ts”

### GUI

mr: [https://gitlab.com/shardeum/validator/gui/-/merge_requests/59](https://gitlab.com/shardeum/validator/gui/-/merge_requests/59)

1. "**/api/**/*.ts"*
2. *"**/hooks/***\*\**/*.ts"
3. "**/utils/**/\*.ts"

### CLI

mr: [https://gitlab.com/shardeum/validator/cli/-/merge_requests/66](https://gitlab.com/shardeum/validator/cli/-/merge_requests/66)

done

### Archiver Server (NOTE: THis duplicated a MR from daniel so I closed this one.)

mr: [https://gitlab.com/shardus/archive/archive-server/-/merge_requests/62](https://gitlab.com/shardus/archive/archive-server/-/merge_requests/62)

 1. "**/dbstore/**/*.ts",*
 2. *"**/utils/**/*.ts",
 3. "**/test/**/*.ts",*
 4. *"**/statistics/index.ts",***
 5. ***"server.ts",***
 6. ***"/profiler/**/*.ts",
 7. "utils.ts",
 8. "**/archivedCycle/**/*.ts",*
 9. *"**/Data/**/*.ts",
10. "State.ts",
11. "P2P.ts",
12. "NodeList.ts",
13. "Logger.ts",
14. "Config.ts",
15. "Crypto.ts",
16. "module-declarations.d.ts",
17. "saveConsoleOutput.ts",
18. "shardus-crypto-types.ts"

### Validator

1. 5 warnings that have slipped in since the linting work was completed

### Shardus-net (rust)

1. src/shardus_net_listener.rs

### Core

 1. "shardus-types.ts",
 2. "Wrapper.ts",
 3. "Utils.ts",
 4. "Template.ts",
 5. "Sync.ts",
 6. "SafetyMode.ts",
 7. "Rotation.ts",
 8. "Refresh.ts",
 9. "NodeList.ts",
10. "Lost.ts",
11. "Join.ts",
12. "GlobalAccounts.ts",
13. "CycleCreator.ts",
14. "CycleChain.ts",
15. "CycleAutoScale.ts",
16. "Context.ts",
17. "Comms.ts",
18. "Archivers.ts",
19. "Apoptosis.ts",
20. "Active.ts",
21. "debugMiddleware.ts",
22. "debug.ts",
23. "config.test.ts",
24. "computePowGenerator.ts",
25. "\*\*\*\*/storage/index.ts",\*\*
26. \*\*"\*\*\*\*/statistics/index.ts",
27. "\*\*\*\*/state-manager/index.ts",\*\*
28. \*\*"\*\*\*\*/snapshot/index.ts",
29. "\*\*\*\*/shardus/index.ts",\*\*
30. \*\*"\*\*\*\*/reporter/index.ts",
31. "\*\*\*\*/rate-limiting/index.ts",\*\*
32. \*\*"\*\*\*\*/random/index.ts",
33. "\*\*\*\*/network/index.ts",\*\*
34. \*\*"\*\*\*\*/logger/index.ts",
35. "\*\*\*\*/load-detection/index.ts",\*\*
36. \*\*"\*\*\*\*/src/index.ts",
37. "\*\*\*\*/http/index.ts",\*\*
38. \*\*"\*\*\*\*/exit-handler/index.ts"

**Comment by chrischabot** [comment]
field | value
|-|-|
url | https://linear.app/shm/issue/SHM-2054/investigate-generate-list-of-files-that-need-linting-fix-so-we-know#comment-715291e2
author | @
date | Fri Aug  4 05:52:20 BST 2023

![image.png](https://uploads.linear.app/1065affa-aebb-4031-b7e5-3fb32c4187d8/1e1d3383-ea9a-4aac-87d4-230dfc01a9cb/29a8372d-9b5b-4657-97c8-549ab7f92399)

**Comment by chrischabot** [comment]
field | value
|-|-|
url | https://linear.app/shm/issue/SHM-2054/investigate-generate-list-of-files-that-need-linting-fix-so-we-know#comment-a79836a5
author | @
date | Sat Nov 11 01:27:41 GMT 2023

assigned this to mehdi just now as he generated the tasks
